src/key.rs

```diff
@@ -916,27 +917,72 @@ impl XOnlyPublicKey {
     }
 }
 
-/// Opaque type used to hold the parity passed between FFI function calls.
+/// Represents the parity passed between FFI function calls.
 #[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
-pub struct Parity(i32);
+pub enum Parity {
+    /// Even parity.
+    Even = 0,
+    /// Odd parity.
+    Odd = 1,
+}
+
+impl Parity {
+    /// Converts parity into a integer (byte) value.
+    pub fn to_u8(self) -> u8 {
+        self as u8
+    }
+
+    /// Converts parity into a integer value.
+    pub fn to_i32(self) -> i32 {
+        self as i32
+    }
+
+    /// Constructs a [`Parity`] from a byte.
+    pub fn from_u8(parity: u8) -> Result<Parity, Error> {
+        Parity::from_i32(parity as i32)
+    }
+
+    /// Constructs a [`Parity`] from a signed integer.
+    pub fn from_i32(parity: i32) -> Result<Parity, Error> {
+        match parity {
+            0 => Ok(Parity::Even),
+            1 => Ok(Parity::Odd),
+            _ => Err(Error::InvalidParityValue),
+        }
+    }
+}
 
 impl From<i32> for Parity {
+    /// Please note, this method is deprecated and will be removed in an upcoming release, it
+    /// is not equivalent to `from_u32()`, it is better to use `Parity::from_u32`.
     fn from(parity: i32) -> Parity {
-        Parity(parity)
+        if parity % 2 == 0 {
+            Parity::Even
+        } else {
+            Parity::Odd
+        }
     }
 }
 
 impl From<Parity> for i32 {
     fn from(parity: Parity) -> i32 {
-        parity.0
+        parity.to_i32()
+    }
+}
+
+impl BitXor for Parity {
+    type Output = Parity;
+
+    fn bitxor(self, rhs: Parity) -> Self::Output {
+        Parity::from_u8(self.to_u8() ^ rhs.to_u8()).expect("XOR result is always a valid parity value")
```

Comment:
I once asked myself: Why do languages have a bitwise XOR operator but no logical XOR operator? The answer is that have one, it's `!=`.

```suggestion
    fn bitxor(self, rhs: Parity) -> Self::Output {
        self != rhs
```

Much simpler but the current code is of course also correct.

Reply:
Cool, I had never thought of this. This only works in code if parity was an int and the language used 1 for true and 0 for false (i.e., C code). In Rust we'd have to map from true/false -> `Parity`. We can do this but I think every dev that reads the code is going to have to scribble down a truth table for XOR to convince themselves that this is the case (well I did anyways).
```
/// This works because Parity has only two values (i.e. only 1 bit of information).
if self == rhs {
    Parity::Even
} else {
    Parity::Odd
}
```

Reply:
I'm unsure right now which to use?

Reply:
Yeah, and of course we'd need your version because mine returned bool and not Parity. (In fact we reimplemented bool and that's not surprising: there are not many meaningful types with exactly two members.)

 I still think the version with the if is more elegant than the original one. But sorry for starting this bikeshedding...

Reply:
I used the if version with some additional comments, cheers.

Reply:
FWIW both compile down to a simple `xor` instruction: https://godbolt.org/z/5WzYd8WoY

Reply:
Wow, I'm amazed at that assembler, all that high level language gone down to just three instructions - that's cool.

Reply:
God bless Zero Cost Abstractions :)